pkgs/development/interpreters/lua-5/interpreter.nix

```diff
@@ -132,6 +133,9 @@ stdenv.mkDerivation rec {
     ln -s "$out/lib/pkgconfig/lua.pc" "$out/lib/pkgconfig/lua-${luaversion}.pc"
     ln -s "$out/lib/pkgconfig/lua.pc" "$out/lib/pkgconfig/lua${luaversion}.pc"
     ln -s "$out/lib/pkgconfig/lua.pc" "$out/lib/pkgconfig/lua${lib.replaceStrings [ "." ] [ "" ] luaversion}.pc"
+
+    # Make documentation outputs of different versions co-installable.
+    mv $out/share/doc/lua $out/share/doc/lua-$version
```

Comment:
Should this be linked to lua/ and lua-version/ like the pkgconfig files above?

Substitution looks like it should be ${version} not $version, did you test if the outputs ended up in the right place?

Reply:
I don't think linking `lua -> lua-$version` is desirable. The whole point of `lua-$version` so documentation is co-installable.

Yes, `$version` works fine. Furthermore, this way it plays nice with `overrideAttrs`.

Reply:
```
>> tree (nix-build -A lua.doc)
/nix/store/8601q2nw98m9wa3m8wzjbvw7dkandk5f-lua-5.2.4-doc
└── share
    └── doc
        └── lua-5.2.4
            ├── contents.html
            ├── logo.gif
            ├── lua.css
            ├── manual.css
            ├── manual.html
            ├── osi-certified-72x60.png
            └── readme.html

3 directories, 7 files

```

Reply:
Linking to both is desirable if anything already depends on it existing at the lua/ path.

I'm not sure how to check if anything does.

Reply:
Technically, if there is something around that referenced in Nix to `${lua}/share/doc/lua/manua.html`, new output already broke it.

On other hand, I introduced `doc` output in dozen of packages already, and nobody complained about things like that, fwiw.